File: AD-HOC/1228/main.dart
import 'dart:io';

void main() {
  String? n;
  while ((n = stdin.readLineSync()) != null) {
    List<int> start = stdin.readLineSync()!.split(' ').map(int.parse).toList();
    List<int> finish = stdin.readLineSync()!.split(' ').map(int.parse).toList();
    int overtakes = 0;
    for (int position = 0; position < int.parse(n!); position++) {
      if (finish[position] != start[position]) {
        int before = start.indexOf(finish[position]);
        start.insert(position, start.removeAt(before));
        overtakes += before - position;
      }
    }
    print(overtakes);
  }
}

File: AD-HOC/1228/main.js
let input = require('fs').readFileSync('/dev/stdin', 'utf-8');
let lines = input.split('\n');
let i = 0
while (lines[i]) {
    let n = parseInt(lines[i++]);
    let start = lines[i++].split(' ').map(Number);
    let finish = lines[i++].split(' ').map(Number);
    let overtakes = 0;
    for (let position in finish) {
        if (finish[position] != start[position]) {
            let before = start.indexOf(finish[position]);
            start.splice(position, 0, start.splice(before, 1)[0]);
            overtakes += before - position;
        }
    }
    console.log(overtakes);
}
